tests/testPsfDetermination.py

```diff
@@ -472,6 +473,19 @@ def testRejectBlends(self):
         self.assertEqual(str(cm.exception), "All PSF candidates removed as blends")
 
 
+class PsfCandidateTestCase(lsst.utils.tests.TestCase):
+    def testNumToReject(self):
+        """Reject the correct number of PSF candidates on each iteration"""
+        # Numerical values correspond to the problem case identified in
+        # DM-8030.
+
+        nCand = 5
+        nIter = 3
+
+        for it, value in ((0, 1), (1, 3), (2, 5)):
```

Comment:
The variables in this test would be a little easier to follow if they used the same names as those in your nice new numCandidatesToReject function (i.e., numBadCandidates, numIter, totalIter).